shared/filestore/s3store.go

```diff
@@ -36,13 +36,18 @@ type S3FileBackend struct {
 	client     *s3.Client
 }
 
+type S3FileBackendAuthError struct {
+	DetailedError string
```

Comment:
What's the reason this needs a `DetailedError` field?

Reply:
The `DetailedError` field is used to determine which operation caused the call to S3 to fail (see the `TestConnection` function). This is useful information because it tells the site admin exactly which AWS permission they are missing.